hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/docker/DockerInspectorManager.java

```diff
@@ -42,171 +44,186 @@
+        DockerInspectorInfo info = null;
+        if (dockerInspectorJar != null) {
+            info = new DockerInspectorInfo(dockerInspectorJar);
+        }
+        return info;
+    }
+
+    private List<File> getAirGapInspectorImageTarfiles() {
+        List<File> airGapInspectorImageTarfiles;
+        airGapInspectorImageTarfiles = new ArrayList<>();
+        final String dockerInspectorAirGapPath = detectConfiguration.getProperty(DetectProperty.DETECT_DOCKER_INSPECTOR_AIR_GAP_PATH);
+        for (final String inspectorName : inspectorNames) {
+            final File osImage = new File(dockerInspectorAirGapPath, IMAGE_INSPECTOR_FAMILY + "-" + inspectorName + ".tar");
+            airGapInspectorImageTarfiles.add(osImage);
+        }
+        return airGapInspectorImageTarfiles;
+    }
+
+    private String getJarFilename(final String version) {
+        final String jarFilename = String.format("blackduck-docker-inspector-%s.jar", version);
+        return jarFilename;
+    }
+
+    private File getConfiguredJar() {
+        logger.debug("Checking for user-specified disk-resident docker inspector jar file");
+        File providedJar = null;
+        final String providedJarPath = detectConfiguration.getProperty(DetectProperty.DETECT_DOCKER_INSPECTOR_PATH);
+        if (StringUtils.isNotBlank(providedJarPath)) {
+            logger.debug(String.format("Using user-provided docker inspector jar path: %s", providedJarPath));
+            final File providedJarCandidate = new File(providedJarPath);
+            if (providedJarCandidate.isFile()) {
+                logger.debug(String.format("Found user-specified jar: %s", providedJarCandidate.getAbsolutePath()));
+                providedJar = providedJarCandidate;
             }
+        }
+        return providedJar;
+    }
+
+    private File getAirGapJar() {
+        final String airGapDirPath = detectConfiguration.getProperty(DetectProperty.DETECT_DOCKER_INSPECTOR_AIR_GAP_PATH);
+        logger.debug(String.format("Checking for air gap docker inspector jar file in %s", airGapDirPath));
+        try {
+            final File airGapJarFile = detectFileFinder.findFilesToDepth(airGapDirPath, "*.jar", 1).get(0);
+            logger.debug(String.format("Found air gap jar: %s", airGapJarFile.getAbsolutePath()));
+            return airGapJarFile;
         } catch (final Exception e) {
-            throw new DetectUserFriendlyException("Unable to find docker inspector version.", e, ExitCodeType.FAILURE_CONFIGURATION);
+            logger.debug(String.format("Did not find a docker inspector jar file in the airgap dir %s", airGapDirPath));
+            return null;
         }
     }
 
-    private String getResponseFromDockerInspector(final String bashExecutablePath, final File dockerInspectorShellScript, final String dockerInspectorArg) throws IOException, ExecutableRunnerException {
+    private File getDownloadedJar() throws DetectUserFriendlyException {
```

Comment:
This method name could be clearer. It's not obvious it could do the actual download work. 

And although not universally followed (and the whole team is not in agreement, cough Eric) I tend to reserve get for only *getting* immutable state and not doing work. I'd call this 'findOrDownloadJar' for example.